agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/git_properties/GitMultiProjectPropertiesLocator.java

```diff
@@ -87,6 +87,12 @@ void searchFile(File file, boolean isJarFile) {
 							GitPropertiesLocatorUtils.GIT_PROPERTIES_TEAMSCALE_COMMIT_TIME);
 					continue;
 				}
+				if (uploader.projectAndCommitAlreadyRegistered(projectAndCommit)) {
+					logger.debug(
+							"Project and commit in git.properties file {} are already registered as upload target. Coverage will not be uploaded multiple times to the same project {} and commit info {}.",
+							file, projectAndCommit.getProject(), projectAndCommit.getCommitInfo());
+					continue;
+				}
```

Comment:
IMHO this deduplication should be an implementation detail of the uploader (i.e. happen in `setTeamscaleProjectForRevision`). Would also make sense to rename it to `addTeamscaleProjectAndCommit` or similar